config/methods.php

```diff
@@ -213,7 +215,15 @@
          * @return Kirby\Cms\Structure
          */
         'toStructure' => function (Field $field) {
-            return new Structure(Yaml::decode($field->value), $field->parent());
+            try {
+                return new Structure(Yaml::decode($field->value), $field->parent());
+            } catch (Exception $e) {
+                if ($field->parent() === null) {
```

Comment:
This is really good! I'd suggest to change the if clause just a little bit to make it more readable. 

```php
if ($field->parent() === null) {
    $message = 'Invalid structure data for "' . $field->key() . '" field';
} else {
    $message = 'Invalid structure data for "' . $field->key() . '" field on parent "' . $field->parent()->title() . '"';
}

throw new InvalidArgumentException($message);
```